### PR TITLE
initial attempt at adding a gradle platform to control dependencies across all the modules we have

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>

### CODE REVIEW

File: modules/framework/galasa-parent/buildSrc/src/main/groovy/galasa.framework.gradle [hunk 1]
```diff
@@ -4,13 +4,15 @@ plugins {
 }
 
 dependencies {
-    implementation 'commons-logging:commons-logging:1.2'
-    implementation 'org.osgi:org.osgi.core:6.0.0'
-    implementation 'org.osgi:org.osgi.service.component.annotations:1.3.0'
-    implementation 'javax.validation:validation-api:2.0.1.Final'
+    implementation platform('dev.galasa:dev.galasa.platform:0.38.0')
```
Comment (techcobweb) [inline modules/framework/galasa-parent/buildSrc/src/main/groovy/galasa.framework.gradle:7]: Could the version here be a variable ?
Like put a variable into the extended context from the root project:
```groovy
ext {
  version = "0.38.0"
}
```
Then access it from here in the plugin ?

Just thinking that if we could, then it's fewer places we need to update with a version number from the set-version.sh script.

If we can't then the set-version.sh script needs updating to do it for us using string substitution. Less elegant than getting it from the top-level script.

File: modules/framework/galasa-parent/dev.galasa.framework.api.cps/build.gradle [hunk 2]
```diff
@@ -5,14 +5,16 @@ plugins {
...
 
 dependencies {
     implementation project(':dev.galasa.framework')
     implementation project(':dev.galasa.framework.api.beans')
     implementation project(':dev.galasa.framework.api.common')
     testImplementation(testFixtures(project(':dev.galasa.framework.api.common')))
-    testImplementation 'org.assertj:assertj-core:3.23.1'
+    testImplementation platform('dev.galasa:dev.galasa.platform:0.38.0')
+
+    testImplementation 'org.assertj:assertj-core:3.23.1' // Overriding version 3.16.1 suggested by the Platform.
```
Comment (techcobweb) [inline modules/framework/galasa-parent/dev.galasa.framework.api.cps/build.gradle]: why are we overriding the one suggested by the platform ?

Comment (techcobweb) [reply modules/framework/galasa-parent/dev.galasa.framework.api.cps/build.gradle]: Seems we need a story to uplift to the 3.23.1 version, to chase-up this loose end.

File: modules/framework/galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd [hunk 3]
```diff
@@ -21,30 +21,30 @@ Import-Package: dev.galasa.framework,\
...
+    commons-codec-*.jar; lib:=true,\
+    commons-io-*.jar; lib:=true,\
+    commons-compress-*.jar; lib:=true,\
+    commons-lang3-*.jar; lib:=true,\
+    commons-collections4-*.jar; lib:=true,\
     jose4j-0.7.12.jar; lib:=true,\
     slf4j-api-1.7.36.jar; lib:=true,\
     protobuf-java-3.21.2.jar; lib:=true,\
     jsr305-3.0.2.jar; lib:=true,\
     swagger-annotations-1.6.6.jar; lib:=true,\
```
Comment (techcobweb) [inline modules/framework/galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd:39]: Wondering why not all these other versions are not also captured in the platform ?

File: modules/framework/galasa-parent/galasa-boot/build.gradle [hunk 4]
```diff
@@ -17,6 +17,30 @@ configurations {
...
 
+dependencies {
+
+    implementation project (':dev.galasa.framework.maven.repository')
+    embedImplementation project (':dev.galasa.framework.maven.repository.spi')
+
+    implementation platform ('dev.galasa:dev.galasa.platform:0.38.0')
+    implementation 'org.apache.felix:org.apache.felix.scr'
+
+    embedImplementation 'org.apache.felix:org.apache.felix.framework:7.0.5'
```
Comment (techcobweb) [inline modules/framework/galasa-parent/galasa-boot/build.gradle:28]: What was the problem with these ones ? We need a story/issue/task to chase these up also ?

File: modules/platform/dev.galasa.platform/build.gradle [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    publications {
+        myPlatform(MavenPublication) {
+            from components.javaPlatform
+
+            groupId = 'dev.galasa'
+            artifactId = 'dev.galasa.platform'
+            version = '0.38.0'
+        }
+
+        // mavenBom(MavenPublication) {
```
Comment (techcobweb) [inline modules/platform/dev.galasa.platform/build.gradle:104]: Need a task on the story to publish the maven bom as an artifact, as we need that for the OBR project I think.

Comment (jadecarino) [reply modules/platform/dev.galasa.platform/build.gradle:104]: Task added to the original [story](https://github.com/galasa-dev/projectmanagement/issues/2052) to complete in another change set

File: modules/framework/galasa-parent/galasa-boot/build.gradle (hunk 4 above)
Comment (jadecarino) [reply modules/framework/galasa-parent/galasa-boot/build.gradle:28]: Task added to the original [story](https://github.com/galasa-dev/projectmanagement/issues/2052) to complete in another change set

File: modules/framework/galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd (hunk 3 above)
Comment (jadecarino) [reply modules/framework/galasa-parent/dev.galasa.framework.k8s.controller/bnd.bnd:39]: Task added to the original [story](https://github.com/galasa-dev/projectmanagement/issues/2052) to complete in another change set

File: modules/framework/galasa-parent/buildSrc/src/main/groovy/galasa.framework.gradle (hunk 1 above)
Comment (jadecarino) [reply modules/framework/galasa-parent/buildSrc/src/main/groovy/galasa.framework.gradle:7]: Task added to the original [story](https://github.com/galasa-dev/projectmanagement/issues/2052) to complete in another change set

File: modules/framework/galasa-parent/dev.galasa.framework.api.cps/build.gradle (hunk 2 above)
Comment (jadecarino) [reply modules/framework/galasa-parent/dev.galasa.framework.api.cps/build.gradle]: Comment expanded to explain why we haven't used 3.23.1 in the platform and why it's overridden here. Raising a story for the above.

**Review by jadecarino — APPROVED** [review]

**Pull request merged**